### ISSUE TITLE
Option to link statically against `libnvrtc_static.a`.

### ISSUE THREAD

**Issue by chr1sj0nes** [NVIDIA/cuda-python#13]
It would be great if it was possible to link against NVRTC statically.

**Comment by leofang** [comment]
Sorry for lack of response @chr1sj0nes. Your ask implies you want to build cuda-python from source yourself. May I ask why?

**Comment by chr1sj0nes** [comment]
Within Google we build everything from source.

**Comment by leofang** [comment]
> Within Google we build everything from source.

Yes, so I was told 🙂  But, I was also told that Google maintains a fork of the whole universe internally (and patch them as needed). Since NVRTC APIs are dlopen'd, it should be possible to patch setup.py with appropriate flags in your internal fork.

Currently we're hesitant to support this feature because of two reasons:
1. We rely on public PyPI to deliver software, and we must do everything we can to avoid generating over-sized wheels. Static linking simply does not play well with the constraints we're facing.
2. We do not have enough bandwidth to develop/maintain/test this additional option (and because of 1 we would basically treat this option as nonexistent)

**Comment by cpcloud** [comment]
Is this something that we can close out? Seems like there's not a lot of demand here/tradeoff curve that's worth it.

**Comment by aterrel** [comment]
@cpcloud please don't close yet.

**Comment by cpcloud** [comment]
Sounds good, I couldn't even if I wanted to, doesn't look like I have permissions!

**Comment by rparolin** [comment]
@aterrel @cpcloud Did we still want to hold onto this issue? Or can we close it?

**Comment by aterrel** [comment]
agree we can close it. I haven't had strong follow up for the use case and we can just open another ticket if it comes up.
